instrumentation/jms/src/main/java/brave/jms/MessageParser.java

```diff
@@ -36,16 +36,35 @@
   }
 
   @Nullable static String channelKind(@Nullable Destination destination) {
-    if (destination instanceof Queue) return "queue";
-    if (destination instanceof Topic) return "topic";
-    return null;
+    if (destination == null) return null;
+    return isQueue(destination) ? "queue" : "topic";
+  }
+
+  /**
+   * Handles special case of a destination being both a Queue and a Topic by checking if the
+   * {@linkplain Queue#getQueueName() queue name} is readable.
+   */
+  static boolean isQueue(@Nullable Destination destination) {
+    boolean isQueue = destination instanceof Queue;
+    boolean isTopic = destination instanceof Topic;
+    if (isQueue && isTopic) {
+      try {
+        isQueue = ((Queue) destination).getQueueName() != null;
+      } catch (Throwable t) {
+        propagateIfFatal(t);
+        log(t, "error getting destination name from {0}", destination, null);
```

Comment:
@anuraaga ex we catch this. we had to recently add these catches everywhere because implementations are free to throw for dumb reasons.